build-system/tasks/visual-diff/index.js

```diff
@@ -862,11 +810,11 @@ async function cleanup_() {
   if (browser_) {
     await browser_.close();
   }
-  if (webServerProcess_ && !webServerProcess_.killed) {
-    // Explicitly exit the webserver.
-    webServerProcess_.kill('SIGKILL');
-    // The child node process has an asynchronous stdout. See #10409.
-    await sleep(100);
+  for (const subprocess of [percyAgentProcess_, webServerProcess_]) {
+    if (subprocess && !subprocess.killed) {
+      // Explicitly exit the processes by "Ctrl+C"-ing them.
+      await subprocess.kill('SIGINT');
```

Comment:
I noticed that the build timed out, not sure what's at play.

![image](https://user-images.githubusercontent.com/26553114/67249711-a4b13900-f436-11e9-9d87-e95cd5325083.png)

Sanity check:
- Is is possible for the `subprocess.kill` to stall?
- If so, I assume the build will fail because Percy does not finalize the build until the agent is stopped?
- Finally, does it make sense to cap the time for which this can run before killing it and failing the overall task?

Reply:
I'm not sure what's happening here, it works on my workstation so this is gonna be difficult to debug. I'll explore these questions

Reply:
Looks like this is related to how Travis and `/bin/sh` work together, according to https://github.com/travis-ci/travis-ci/issues/704#issuecomment-328278149
Replacing the default shell to `/bin/bash` fixes the issue. There _could_ be some side effects to this with other tasks. PTAL at the recent changes and tell me what you think

Reply:
Nice sleuthing. I sanity tested your branch on a Macbook and everything looks good so far.